client.go

```diff
@@ -616,6 +618,36 @@ func NewClient[TTx any](driver riverdriver.Driver[TTx], config *Config) (*Client
 	return client, nil
 }
 
+func (c *Client[TTx]) AddQueue(queueName string, queueConfig QueueConfig) {
+	c.producersByQueueNameMu.Lock()
+	defer c.producersByQueueNameMu.Unlock()
+	c.producersByQueueName[queueName] = newProducer(&c.baseService.Archetype, c.driver.GetExecutor(), &producerConfig{
+		ClientID:          c.config.ID,
+		Completer:         c.completer,
+		ErrorHandler:      c.config.ErrorHandler,
+		FetchCooldown:     c.config.FetchCooldown,
+		FetchPollInterval: c.config.FetchPollInterval,
+		JobTimeout:        c.config.JobTimeout,
+		MaxWorkers:        queueConfig.MaxWorkers,
+		Notifier:          c.notifier,
+		Queue:             queueName,
+		RetryPolicy:       c.config.RetryPolicy,
+		SchedulerInterval: c.config.schedulerInterval,
+		StatusFunc:        c.monitor.SetProducerStatus,
+		Workers:           c.config.Workers,
+	})
+	c.monitor.InitializeProducerStatus(queueName)
+}
+
+func (c *Client[TTx]) RemoveQueue(queueName string) {
+	c.producersByQueueNameMu.Lock()
+	defer c.producersByQueueNameMu.Unlock()
+	delete(c.producersByQueueName, queueName)
+
+	// Remove queue from currentSnapshot.Producers
+	c.monitor.RemoveProducerStatus(queueName)
```

Comment:
Safe removal is a bit trickier than this. We need to actually shut down the producer somehow, and then the producer itself should be responsible for reporting its status to the monitor until it is shut down. At that point, yeah, we need to somehow remove it from the monitor's map so we don't keep old state around in that map forever.

Maybe `SetProducerStatus` should just be smart enough to do that last part automatically when the producer is stopped?

@brandur thoughts on how to safely trigger the single producer to shut down? And is this API suitable for an operation which may take awhile (until all running jobs go away)? Keeping in mind there is no obvious way to cancel the context just for a single producer, and we may also need to differentiate between `Stop` and `StopAndCancel` for each individual producer—otherwise how will the producer know when it needs to cancel the context and try to more aggressively shut down running jobs?

In either case we still need to somehow wait for clean shutdown to complete, which may actually never happen before the program exits if the running jobs never exit.

Reply:
Yeah definitely pretty tricky. I find the existing `StopAndCancel` to be quite confusing to reason about, so it'd be nice if we could avoid adding another version of that here.

One possibility might be that `RemoveQueue` will only wait for a graceful shutdown, and if a hard shutdown is necessary, the caller will just have to use `StopAndCancel`.

Also, if it's not needed, I like your idea of just punting on `RemoveQueue` for now.

_Or_, a possible variation of that is to have `RemoveQueue`, but only allow it to be called if the client is stopped, which would allow it if necessary, but make a lot of the harder problems go away. If we went that route though, we should make sure to future-proof the API (e.g. by making it take a `ctx` if we think there will be a hard stop later) in case we want to remove this limitation in the future.